src/calls/components/Calendar.jsx

```diff
@@ -0,0 +1,314 @@
+
+const months = [
+    'Jan',
+    'Feb',
+    'Mar',
+    'Apr',
+    'Maj',
+    'Jun',
+    'Jul',
+    'Avg',
+    'Sep',
+    'Okt',
+    'Nov',
+    'Dec'
+];
+
+class Calendar extends Component {
+    state = {
+        selectedDate: new Date()
+    };
+
+    componentDidMount() {
+        this.props.onDateSelect(this.state.selectedDate);
+    }
+
+    handleOnDateClick = event => {
+        const clickedDay = event.target.textContent;
+
+        const type = event.target.getAttribute('type');
+        const { selectedDate } = this.state;
+
+        if (type === 'prev') {
+            this.setState(
+                {
+                    selectedDate: setDate(
+                        subMonths(selectedDate, 1),
+                        clickedDay
+                    )
+                },
+                () => this.props.onDateSelect(this.state.selectedDate)
+            );
+        } else if (type === 'next') {
+            this.setState(
+                {
+                    selectedDate: setDate(
+                        addMonths(selectedDate, 1),
+                        clickedDay
+                    )
+                },
+                () => this.props.onDateSelect(this.state.selectedDate)
+            );
+        } else {
+            this.setState(
+                {
+                    selectedDate: addDays(
+                        selectedDate,
+                        clickedDay - selectedDate.getDate()
+                    )
+                },
+                () => this.props.onDateSelect(this.state.selectedDate)
+            );
+        }
+    };
+
+    handleMonthForward = event => {
+        const { selectedDate } = this.state;
+        this.setState({ selectedDate: addMonths(selectedDate, 1) }, () =>
+            this.props.onDateSelect(this.state.selectedDate)
+        );
+    };
+
+    handleMonthBackward = event => {
+        const { selectedDate } = this.state;
+        this.setState({ selectedDate: subMonths(selectedDate, 1) }, () =>
+            this.props.onDateSelect(this.state.selectedDate)
+        );
+    };
+
+    handleChangeDate = event => {
+        const option = event.target.textContent;
+        const { selectedDate } = this.state;
+
+        if (!parseInt(option)) {
+            this.setState(
+                {
+                    selectedDate: setMonth(selectedDate, months.indexOf(option))
+                },
+                () => this.props.onDateSelect(this.state.selectedDate)
+            );
+        } else {
+            this.setState(
+                {
+                    selectedDate: setYear(selectedDate, option)
+                },
+                () => this.props.onDateSelect(this.state.selectedDate)
+            );
+        }
+    };
+
+    render() {
+        const numOfDays = getDaysInMonth(this.state.selectedDate);
+        const startDay = getDay(startOfMonth(this.state.selectedDate));
+        const endDay = getDay(endOfMonth(this.state.selectedDate));
```

Comment:
Grab the selected date at the top of the function. Accessing the same property of the same object multiple times is wasteful.